### DISCUSSION TITLE
How to generate keycloack token without a record in keycloack

### DISCUSSION THREAD

**Discussion by shiva-rakshith** [Sunbird-RC/community#166] [Issues]
Reference: #163, we have two schemas `Organization` and `User`, Initially organization record is created by a registry admin and later user will be created and linked to that organization. As organization won't have record in keycloack, how can we generate keycloack token to make updates to organization record? (or) As users will be linked to organization entity, can we make updates to organization record using user keycloack token?

cc: @vrayulu @maheshkumargangula

**Comment by pramodkvarma** [comment]
User who is assigned the role of org admin should be able to update org details in registry without having an org entry in keycloak. That's my understanding.

 @dileepbapat ?

**Reply by dileepbapat** [reply]
Yes, Organization when ownership attributes are mapped SunbirdRC will set the owner based on email/mobile that is specified in the Organization. Thereon particular organization will be managed by login associated with the email / phone number listed in the Organization.

this is handled by rule in : https://github.com/Sunbird-RC/sunbird-rc-core/blob/988513204f7ac0a624dbbf10ebaefb84ed25ffd3/java/middleware/registry-middleware/workflow/src/main/resources/workflow/statetransitions.drl#L43

**Reply by shiva-rakshith** [reply]
@dileepbapat - Yes, defining ownership attributes in `Organization` schema will create an entry in keycloack and we can make updates to organization record, this is one way of token generation and making updates. But we are intending to do the updates to organization record through the user entity as both of them are linked. How can we achieve this?
